src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -568,5 +580,17 @@ private void CheckSize(int countToAdd)
                 }
             }
         }
+
+        private struct EncodingPreamblePair
+        {
+            public readonly Encoding Encoding;
+            public readonly byte[] Preamble;
+
+            public EncodingPreamblePair(Encoding encoding, byte[] preamble)
```

Comment:
The only place you're constructing these, you're passing in SomeEncoding and SomeEncoding.GetPreamble().  You could simplify those call sites by just making the constructor be:
```C#
public EncodingPreamblePair(Encoding encoding)
{
    Encoding = encoding;
    Preamble = encoding.GetPreamble();
}
```
or you could add a constructor that does that:
```C#
public EncodingPreamblePair(Encoding encoding) : this(encoding, encoding.GetPreamble())
{
}

public EncodingPreamblePair(Encoding encoding, byte[] preamble)
{
    Encoding = encoding;
    Preamble = preamble;
}
```